src/aznfswatchdog

```diff
@@ -0,0 +1,190 @@
+    #
+    # where the format is
+    # <blobendpoint>:/account/container LOCAL_IP blobendpoint_ip
+    # 
+    IFS=$'\n' lines=$(cat $MOUNTMAP) 
+
+    for line in $lines; do
+        if [ -z "$line" ]; then
+            continue
+        fi
+
+        l_share=$(echo "$line" | awk '{print $1}')
+        l_host=$(echo "$l_share" | cut -d: -f1)
+        l_dir=$(echo "$l_share" | cut -d: -f2)
+        l_ip=$(echo "$line" | awk '{print $2}')
+        l_nfsip=$(echo "$line" | awk '{print $3}')
+        l_pid=$(echo "$line" | awk '{print $4}')
+
+        if [ -z "$l_host" -o -z "$l_dir" -o -z "$l_ip" -o -z "$l_nfsip" ]; then
+            wecho "[FATAL] Deleting invalid line in $MOUNTMAP: [$line]."
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        # Since we added it to the MOUNTMAP file, it cannot be invalid.
+        if ! is_private_ip "$l_ip"; then
+            wecho "[FATAL] local ip ($l_ip) is invalid!"
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        # Since we added it to the MOUNTMAP file, it cannot be invalid.
+        if ! is_valid_ipv4_address "$l_nfsip" || is_private_ip "$l_nfsip"; then
+            wecho "[FATAL] Blob endpoint ip ($l_nfsip) is invalid!"
+            if ! ensure_mountmap_not_exist "$line"; then
+                eecho "Failed to delete [$line] from MOUNTMAP."
+            fi
+            continue
+        fi
+
+        #
+        # This entry was added by get_free_local_ip() call in mount.aznfs to
+        # to ensure two parallel mount requests do not get same local ip. Skip
+        # this entry if PID is active.
+        #
+        if [ ! -z $l_pid ]; then
+            if ! ps -p $l_pid > /dev/null; then
+                wecho "[FATAL] PID ($l_pid) is not active!"
+                if ! ensure_mountmap_not_exist "$line"; then
+                    eecho "Failed to delete [$line] from MOUNTMAP."
+                fi
+            else
+                continue
+            fi
+        fi
+
+        #
+        # Local share name. Note that we mount the local IP and not the actual
+        # Blob endpoint IP and map local IP to actual Blob endpoint IP using
+        # DNAT rule.
+        #
+        m_share="$l_ip:$l_dir"
+
+        #
+        # Delete entry from MOUNTMAP if share is unmounted.
+        # TODO: What if user unmounts and mounts before MONITOR_INTERVAL_SECS secs?
+        #
+        if ! echo "$findmnt" | grep " nfs $m_share " > /dev/null; then
+            # delete the line from MOUNTMAP file.
+            pecho "Deleting unmounted share from MOUNTMAP [$line]."
+            if ! delete_iptable_entry $l_ip $l_nfsip; then
+                eecho "iptables failed to delete DNAT rule [$l_ip -> $l_nfsip]."
+                exit 1
```

Comment:
we shouldn't exit the service
same for all places where you call exit
it should only exit if it cannot safely continue

In this case it should fall through (after logging) and attempt to delete the mountmap entry.